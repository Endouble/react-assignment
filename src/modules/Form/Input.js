import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { formElement } from '../../styles/mixins';

class Input extends React.Component {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event) {
        const { value } = event.target;
        const { filterBy } = this.props.filter;
        this.props.updateFilter({
            value,
            filterBy,
        });
    }

    render() {
        const { filter } = this.props;
        return <StyledInput placeholder={`filter by: ${filter.filterBy}`} onChange={this.handleChange} />;
    }
}

/*
    Input Styles
*/
const StyledInput = styled.input`
    max-width: 60%;
    ${formElement()};
`;

/*
    Input propTypes
*/
Input.propTypes = {
    updateFilter: PropTypes.func,
    filter: PropTypes.shape({
        filterBy: PropTypes.string,
    }),
};

Input.defaultProps = {
    updateFilter: null,
    filter: {},
};


export default Input;
