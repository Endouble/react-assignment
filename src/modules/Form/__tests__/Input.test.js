import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Input from '../Input';

configure({ adapter: new Adapter() });

const props = {
    filter: {
        filterBy: 'placeholder',
    },
};

describe('Input', () => {
    it('renders correctly', () => {
        const component = shallow(<Input />);
        expect(component).toMatchSnapshot();
    });

    it('has right placeholder copy', () => {
        const component = shallow(<Input {...props} />);
        expect(component.prop('placeholder')).toEqual('filter by: placeholder');
    });
});
