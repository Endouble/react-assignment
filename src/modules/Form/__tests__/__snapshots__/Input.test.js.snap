// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<styled.input
  onChange={[Function]}
  placeholder="filter by: undefined"
/>
`;
